src/ring_theory/noetherian.lean

```diff
@@ -69,6 +69,20 @@ theorem fg_def {N : submodule R M} :
   exact ⟨t, rfl⟩
 end⟩
 
+lemma fg_iff_exists_fin_generating_fam {N : submodule R M} :
+  N.fg ↔ ∃ (n : ℕ) (s : fin n → M), span R (range s) = N :=
+begin
+  classical,
+  rw fg_def,
+  split,
+  { rintros ⟨S, Sfin, hS⟩,
+    rcases Sfin.fin_param with ⟨n, f, -, h⟩,
+    use [n, f],
+    rwa h },
+  { rintros ⟨n, s, hs⟩,
+    refine ⟨range s, finite_range s, hs⟩ },
```

Comment:
Slightly golfed - `classical` wasn't be used at all here, and `fin_param` is just `fin_embedding` plus the extra result you discard:
```suggestion
  rw fg_def,
  split,
  { rintros ⟨S, Sfin, hS⟩,
    obtain ⟨n, f, rfl⟩ := Sfin.fin_embedding,
    exact ⟨n, f, hS⟩, },
  { rintros ⟨n, s, hs⟩,
    refine ⟨range s, finite_range s, hs⟩ },
```